# This file is part of HiGFXback

# requires
REQUIRES="cmake3-build vulkan-headers-build wayland-build"

PKG_CONFIG_PATH=/wl/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

# configure
GENERATOR=${GENERATOR-Unix Makefiles}
PKG_CONFIG_PATH=/wl/lib/pkgconfig cmake -DBUILD_WSI_XCB_SUPPORT=OFF -DBUILD_WSI_XLIB_SUPPORT=OFF -DCMAKE_INSTALL_PREFIX=/wl -B build -G "$GENERATOR"

# build
cmake --build build

# install
install -d $DESTDIR/wl/lib
install build/loader/libvulkan.so.1.* $DESTDIR/wl/lib/libvulkan.so.1
ln -sf libvulkan.so.1 $DESTDIR/wl/lib/libvulkan.so
install -d $DESTDIR/wl/lib/pkgconfig
install -m 644 build/loader/vulkan.pc $DESTDIR/wl/lib/pkgconfig

# build.pc
install -d $DESTDIR/wl/share/pkgconfig
cat > $DESTDIR/wl/share/pkgconfig/vulkan-loader-build.pc << EOF
Name: Vulkan-Loader
Version: 1.2.160
Description: Vulkan Installable Client Driver (ICD) loader
Requires: $REQUIRES

devel=\\
/wl/lib/libvulkan.so \\
/wl/lib/pkgconfig/vulkan.pc

exec=\\
/wl/lib/libvulkan.so.1
EOF

exit
--- Vulkan-Loader-1.2.160.orig/CMakeLists.txt
+++ Vulkan-Loader-1.2.160/CMakeLists.txt
@@ -30,7 +30,7 @@
 add_definitions(-DVK_ENABLE_BETA_EXTENSIONS)
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
-find_package(PythonInterp 3 QUIET)
+find_package(Python3 QUIET)
 
 if(IS_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/external/googletest)
     option(BUILD_TESTS "Build Tests" ON)
@@ -192,9 +192,9 @@
 endif()
 
 # Optional codegen target
-if(PYTHONINTERP_FOUND)
+if(Python3_Interpreter_FOUND)
     add_custom_target(VulkanLoader_generated_source
-                      COMMAND ${PYTHON_EXECUTABLE} ${PROJECT_SOURCE_DIR}/scripts/generate_source.py
+                      COMMAND Python3::Interpreter ${PROJECT_SOURCE_DIR}/scripts/generate_source.py
                               ${VulkanRegistry_DIR} --incremental
                       )
 else()
